samcli/local/docker/lambda_image.py

```diff
@@ -211,7 +211,7 @@ def build(self, runtime, packagetype, image, layers, architecture, stream=None,
                 )
                 image_not_found = True
             else:
-                raise DockerDistributionAPIError("Unknown API error received from docker") from e
+                raise DockerDistributionAPIError("Error: " + str(e)) from e
```

Comment:
I think `str(e)` should be enough here. `Error: ` gets added when the exception is thrown by default.